DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1

```diff
@@ -134,48 +130,162 @@ function ValidateProperties
 
     if(!([System.Net.Ipaddress]::TryParse($ip, [ref]0)))
     {
-       throw "IP Address *$IPAddress* is not in the correct format. Please correct the ipaddress in the configuration and try again"
+        throw ( @(
+            "IP Address *$IPAddress* is not in the correct format. Please correct the ipaddress "
+            "in the configuration and try again"
+            ) -join "" | Out-String
+        )
     }
 
     try
     {
         Write-Verbose -Message "Checking the IPAddress ..."
         #Get the current IP Address based on the parameters given.
-        $currentIP = Get-NetIPAddress -InterfaceAlias $InterfaceAlias -AddressFamily $AddressFamily -ErrorAction Stop
+        $currentIP = Get-NetIPAddress -InterfaceAlias $InterfaceAlias -AddressFamily `
+            $AddressFamily -ErrorAction Stop
+
+        # Build hash table of current settings
+        $currentSettings = @{}
+        $currentSettings['IPAddress'] = $currentIP.IPAddress
+        $currentSettings['PrefixLength'] = $currentIP.PrefixLength
+        $currentSettings['InterfaceAlias'] = $currentIP.InterfaceAlias
+
+        # Flag to signal whether settings are correct
+        $requiresChanges = $false
 
         #Test if the IP Address passed is equal to the current ip address
-        if(!$currentIP.IPAddress.Contains($IPAddress))
+        if(!$currentSettings['IPAddress'].Equals($IPAddress))
         {
-            Write-Verbose -Message "IPAddress not correct. Expected $IPAddress, actual $($currentIP.IPAddress)"
-            $Parameters = @{}
+            Write-Verbose -Message ( @(
+                "IPAddress does NOT match desired state. Expected $IPAddress, actual "
+                "$($currentSettings['IPAddress'])."
+                ) -join "" | Out-String
+            )
+            $requiresChanges = $true
```

Comment:
Ok so this is place where I'm going to start arguing breaking up Test and Set. So if I hit this logic I know I need a change. I'm then going to validate PrefixLength, Default Gateway, and DHCP before I actually exit out of this function. If this was purely a Test-TargetResoruce function we could return here and dump out of the script and begin the correction.